tests/inputs/stmtTests.mg

```diff
@@ -17,4 +17,107 @@ implementation StmtTests = {
         // Invalid assignment
         t1 = mkT2();
     }
-};
+}
+
+// TODO: add more tests
+implementation VarExprTests = {
+
+    require type T;
+    require type U;
+
+    procedure conflicting_type_annotation() = {
+        var x: T;
+        // should succeed
+        x: T;
+        // should fail
+        x: U;
+    }
+}
+
+
+// TODO: add more tests
+implementation CallExprTests = {
+
+    require type T;
+    require type U;
+
+    function overload_return_type(): T;
+    function overload_return_type(): U;
+
+    procedure call_functions() = {
+
+        // should succeed
+        var explicit_T: T = overload_return_type();
+
+        // should succeed
+        var explicit_U: U = overload_return_type();
+
+        // should fail
+        var implicit_type = overload_return_type();
+    }
+}
+implementation LetExprTests = {
+
+    require type T;
+    require type U;
+
+    procedure var_assignment() = {
+        // should succeed
+        var x: T;
+        // should fail
+        var x: U;
+
+    }
+
+    procedure var_declaration_infer_type() = {
+        // should succeed
+        // TODO(mariuslarnoy 2021/07/19): this is not implemented yet
+        var x;
+    }
+}
+implementation IfExprTests = {
+
+    // TODO: Could add more tests to discover compiler errors in
+    // the functionality of mode and type updates
```

Comment:
I'm not sure what the tests should be, it was more of a note to myself. I noticed there were a couple of `CompilerErr` in this part of the typechecking pass. If those were to trigger, I guess it would require a fix anyways.